File: app/src/main/java/com/stevenfrew/beatprompter/filter/TemporarySetListFilter.java
package com.stevenfrew.beatprompter.filter;

import com.stevenfrew.beatprompter.cache.SetListFile;
import com.stevenfrew.beatprompter.cache.SongFile;

import java.util.List;

public class TemporarySetListFilter extends SetListFileFilter
{
    public TemporarySetListFilter(SetListFile setListFile, List<SongFile> songs)
    {
        super(setListFile,songs);
    }
    public void addSong(SongFile sf)
    {
        mSongs.add(sf);
    }
    public boolean equals(Object o) {
        return o!=null && o instanceof TemporarySetListFilter;
    }
    public void clear()
    {
        mMissingSongs.clear();
        mSongs.clear();
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/ui/filter/TemporarySetListFilter.kt
package com.stevenfrew.beatprompter.ui.filter

import com.stevenfrew.beatprompter.cache.SetListFile
import com.stevenfrew.beatprompter.song.SongInfo

class TemporarySetListFilter(
	setListFile: SetListFile,
	songs: List<SongInfo>
) : SetListFileFilter(setListFile, songs) {
	fun addSong(sf: SongInfo) = songs.add(sf to null)
	override fun equals(other: Any?): Boolean = other != null && other is TemporarySetListFilter
	override fun hashCode(): Int = javaClass.hashCode()

	fun clear() {
		mMissingSetListEntries.clear()
		songs.clear()
	}
}
